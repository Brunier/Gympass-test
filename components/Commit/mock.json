{
  "sha": "942a765b53662936679b8119ed4533d972298d2a",
  "commit": {
    "author": {
      "name": "Pedro Bullo Junior",
      "email": "pedro.bj@hotmail.com",
      "date": "2019-01-21T22:59:19Z"
    },
    "committer": {
      "name": "GitHub",
      "email": "noreply@github.com",
      "date": "2019-01-21T22:59:19Z"
    },
    "message": "Workaround empty child",
    "tree": {
      "sha": "81c34b008db2a4fda4df20404a498e0127e4a817",
      "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/trees/81c34b008db2a4fda4df20404a498e0127e4a817"
    },
    "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/commits/942a765b53662936679b8119ed4533d972298d2a",
    "comment_count": 0,
    "verification": {
      "verified": true,
      "reason": "valid"
    }
  },
  "url": "https://github.com/Gympass/react-native-svg-uri/commit/942a765b53662936679b8119ed4533d972298d2a"
}
