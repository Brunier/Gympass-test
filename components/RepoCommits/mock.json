{
  "data" : [
    {
      "sha": "942a765b53662936679b8119ed4533d972298d2a",
      "commit": {
        "author": {
          "name": "Pedro Bullo Junior",
          "email": "pedro.bj@hotmail.com",
          "date": "2019-01-21T22:59:19Z"
        },
        "committer": {
          "name": "GitHub",
          "email": "noreply@github.com",
          "date": "2019-01-21T22:59:19Z"
        },
        "message": "Workaround empty child",
        "tree": {
          "sha": "81c34b008db2a4fda4df20404a498e0127e4a817",
          "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/trees/81c34b008db2a4fda4df20404a498e0127e4a817"
        },
        "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/commits/942a765b53662936679b8119ed4533d972298d2a",
        "comment_count": 0,
        "verification": {
          "verified": true,
          "reason": "valid"
        }
      },
      "url": "https://github.com/Gympass/react-native-svg-uri/commit/942a765b53662936679b8119ed4533d972298d2a"
    },
    {
      "sha": "c6841a6e7d1a6ee5cf5b515d4faea6002e166f3c",
      "commit": {
        "author": {
          "name": "matc4",
          "email": "matias.s.cortes@gmail.com",
          "date": "2018-09-13T12:34:04Z"
        },
        "committer": {
          "name": "GitHub",
          "email": "noreply@github.com",
          "date": "2018-09-13T12:34:04Z"
        },
        "message": "Merge pull request #100 from nesc58/masteFix: Text elements / transform + clipPath attribute added",
        "tree": {
          "sha": "97dae56ead52ab59e12b98848f4e38d89265040f",
          "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/trees/97dae56ead52ab59e12b98848f4e38d89265040f"
        },
        "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/commits/c6841a6e7d1a6ee5cf5b515d4faea6002e166f3c",
        "comment_count": 0,
        "verification": {
          "verified": true,
          "reason": "valid"
        }
      },
      "url": "https://github.com/Gympass/react-native-svg-uri/commit/c6841a6e7d1a6ee5cf5b515d4faea6002e166f3c"
    }
  ]
}
