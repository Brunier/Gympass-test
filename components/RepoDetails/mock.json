{
  "name": "interview-test",
  "description": "Interview test project",
  "url": "https://api.github.com/repos/Gympass/interview-test",
  "commits_url": "https://api.github.com/repos/Gympass/interview-test/commits",
  "created_at": "2018-05-22T19:57:37Z",
  "updated_at": "2019-01-24T02:51:23Z",
  "language": "Javascript",
  "stars": 3,
  "watchers": 3,
  "forks": 4
}
