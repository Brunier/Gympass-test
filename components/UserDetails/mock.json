{
  "repos": [
    {
      "name": "mindbody-api",
      "description": "A Ruby interface to the MindBody API v0.5",
      "url": "https://api.github.com/repos/Gympass/mindbody-api",
      "commits_url": "https://api.github.com/repos/Gympass/mindbody-api/commits",
      "created_at": "2018-04-09T17:04:01Z",
      "updated_at": "2018-04-13T21:52:29Z",
      "language": "Ruby",
      "stars": 0,
      "watchers": 0,
      "forks": 0
    },
    {
      "name": "rails-intl-tel-input",
      "description": "Integrate with intl-tel-input to provide international telephone numbers input and form helper.",
      "url": "https://api.github.com/repos/Gympass/rails-intl-tel-input",
      "commits_url": "https://api.github.com/repos/Gympass/rails-intl-tel-input/commits",
      "created_at": "2017-11-14T17:27:57Z",
      "updated_at": "2018-03-16T12:40:19Z",
      "language": "JavaScript",
      "stars": 0,
      "watchers": 0,
      "forks": 0
    }
  ],
  "name": "gympass",
  "img": "https://avatars3.githubusercontent.com/u/26311162?v=4",
  "loading": false,
  "notFound": false
}
