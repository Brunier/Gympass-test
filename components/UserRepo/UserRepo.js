import React from 'react';
import styled from "styled-components";
import Text from "../Text/Text";
import TextParagraph from "../TextParagraph/TextParagraph";
import TextTitle from "../TextTitle/TextTitle";
import moment from 'moment'
import momentBr from 'moment/locale/pt-br'
moment.locale('momentBr');
const Container = styled.div`
  width: 21%;
  border: 0.13vw solid #00757e;
  margin-right: 2%;
  padding: 1%;
  margin-bottom: 20px;
  cursor: pointer;
  
  &:hover {
    background: #00757e;
    
    span, p {
      color: #fff;
    }
  }
  
  &:nth-child(4n) {
      margin-right: 0 
    }
`;

 class UserRepo extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return(
            <Container onClick={this.props.onClick}>
                <TextTitle padding={"5px 10% 5px 5px"}>{this.props.repo.name}</TextTitle>
                <TextParagraph height={96}>{this.props.repo.description}</TextParagraph>
                <Text bold={true} fontSize={14}>{moment(this.props.repo.updated_at).format('llll')}</Text>
                <Text bold={true} fontSize={14}> - </Text>
                <Text bold={true} fontSize={14}>{this.props.repo.language ? this.props.repo.language : "Nenhuma"}</Text>

            </Container>
        )
    }
}

export default UserRepo;
