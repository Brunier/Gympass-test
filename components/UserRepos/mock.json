{
  "data": [
    {
      "name": "mindbody-api",
      "description": "A Ruby interface to the MindBody API v0.5",
      "url": "https://api.github.com/repos/Gympass/mindbody-api",
      "commits_url": "https://api.github.com/repos/Gympass/mindbody-api/commits",
      "created_at": "2018-04-09T17:04:01Z",
      "updated_at": "2018-04-13T21:52:29Z",
      "language": "Ruby",
      "stars": 0,
      "watchers": 0,
      "forks": 0
    },
    {
      "name": "rails-intl-tel-input",
      "description": "Integrate with intl-tel-input to provide international telephone numbers input and form helper.",
      "url": "https://api.github.com/repos/Gympass/rails-intl-tel-input",
      "commits_url": "https://api.github.com/repos/Gympass/rails-intl-tel-input/commits",
      "created_at": "2017-11-14T17:27:57Z",
      "updated_at": "2018-03-16T12:40:19Z",
      "language": "JavaScript",
      "stars": 0,
      "watchers": 0,
      "forks": 0
    },
    {
      "name": "devise",
      "description": "Flexible authentication solution for Rails with Warden.",
      "url": "https://api.github.com/repos/Gympass/devise",
      "commits_url": "https://api.github.com/repos/Gympass/devise/commits",
      "created_at": "2018-01-17T00:17:46Z",
      "updated_at": "2018-01-17T00:17:52Z",
      "language": "Ruby",
      "stars": 0,
      "watchers": 0,
      "forks": 0
    },
    {
      "name": "AvaTax-REST-V2-Ruby-SDK",
      "description": "Ruby SDK for AvaTax REST v2",
      "url": "https://api.github.com/repos/Gympass/AvaTax-REST-V2-Ruby-SDK",
      "commits_url": "https://api.github.com/repos/Gympass/AvaTax-REST-V2-Ruby-SDK/commits",
      "created_at": "2018-01-09T23:23:04Z",
      "updated_at": "2018-01-09T23:23:06Z",
      "language": "Ruby",
      "stars": 0,
      "watchers": 0,
      "forks": 0
    }
  ]
}
