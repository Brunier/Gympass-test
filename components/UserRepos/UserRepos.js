import React from 'react';
import styled from "styled-components";
import UserRepo from '../UserRepo/UserRepo';

const Container = styled.div`
  margin-top: 40px;
  flex-wrap: wrap;
  display: flex;
  margin-bottom: 60px;
`;

class UserRepos extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return(
            <Container>
                {this.props.repos.map(repo => {
                    return(
                        <UserRepo key={repo.name} repo={repo} onClick={() =>{
                            this.props.repositorySelect(repo)
                        }} />
                    )
                })}
            </Container>
        )
    }
}

export default UserRepos;
