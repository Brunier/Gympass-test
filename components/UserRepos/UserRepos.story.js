import React from 'react';
import { storiesOf } from '@storybook/react';
import UserRepos from './UserRepos';
import mock from "./mock"


storiesOf('UserRepos', module)
    .add('Normal', () => (
        <UserRepos  repos={mock.data}/>
    ))
