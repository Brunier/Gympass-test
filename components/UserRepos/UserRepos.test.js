import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'
import mock from './mock';

// Components
import UserRepos from './UserRepos';
import UserRepo from '../UserRepo/UserRepo';
import moment from "moment/moment";

function setup(props) {
    const wrapper = shallow(<UserRepos {...props} />);
    return { wrapper };
}

const repos = mock.data;

describe('UserSearch Test Suite', () => {
    it('Verificar a quantidade de UserRepo', () => {
        const { wrapper } = setup({repos});
        expect(wrapper.find(UserRepo)).toHaveLength(repos.length);
    });

    it('Snapshot', () => {
        const tree = renderer.create(<UserRepos repos={repos} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
});
