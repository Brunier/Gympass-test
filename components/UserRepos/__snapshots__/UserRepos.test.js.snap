// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSearch Test Suite Snapshot 1`] = `
<div
  className="sc-ifAKCX hIyfxW"
>
  <div
    className="sc-bxivhb hHKjDW"
    onClick={[Function]}
  >
    <h1
      className="sc-bdVaJa eiPDuz"
    >
      mindbody-api
    </h1>
    <p
      className="sc-bwzfXH fRwYcu"
      height={96}
    >
      A Ruby interface to the MindBody API v0.5
    </p>
    <span
      className="sc-htpNat ivUPnB"
      fontSize={14}
    >
       
      Sex, 13 de Abr de 2018 às 18:52
    </span>
    <span
      className="sc-htpNat ivUPnB"
      fontSize={14}
    >
       - 
    </span>
    <span
      className="sc-htpNat ivUPnB"
      fontSize={14}
    >
      Ruby
    </span>
  </div>
  <div
    className="sc-bxivhb hHKjDW"
    onClick={[Function]}
  >
    <h1
      className="sc-bdVaJa eiPDuz"
    >
      rails-intl-tel-input
    </h1>
    <p
      className="sc-bwzfXH fRwYcu"
      height={96}
    >
      Integrate with intl-tel-input to provide international telephone numbers input and form helper.
    </p>
    <span
      className="sc-htpNat ivUPnB"
      fontSize={14}
    >
       
      Sex, 16 de Mar de 2018 às 09:40
    </span>
    <span
      className="sc-htpNat ivUPnB"
      fontSize={14}
    >
       - 
    </span>
    <span
      className="sc-htpNat ivUPnB"
      fontSize={14}
    >
      JavaScript
    </span>
  </div>
  <div
    className="sc-bxivhb hHKjDW"
    onClick={[Function]}
  >
    <h1
      className="sc-bdVaJa eiPDuz"
    >
      devise
    </h1>
    <p
      className="sc-bwzfXH fRwYcu"
      height={96}
    >
      Flexible authentication solution for Rails with Warden.
    </p>
    <span
      className="sc-htpNat ivUPnB"
      fontSize={14}
    >
       
      Ter, 16 de Jan de 2018 às 22:17
    </span>
    <span
      className="sc-htpNat ivUPnB"
      fontSize={14}
    >
       - 
    </span>
    <span
      className="sc-htpNat ivUPnB"
      fontSize={14}
    >
      Ruby
    </span>
  </div>
  <div
    className="sc-bxivhb hHKjDW"
    onClick={[Function]}
  >
    <h1
      className="sc-bdVaJa eiPDuz"
    >
      AvaTax-REST-V2-Ruby-SDK
    </h1>
    <p
      className="sc-bwzfXH fRwYcu"
      height={96}
    >
      Ruby SDK for AvaTax REST v2
    </p>
    <span
      className="sc-htpNat ivUPnB"
      fontSize={14}
    >
       
      Ter, 9 de Jan de 2018 às 21:23
    </span>
    <span
      className="sc-htpNat ivUPnB"
      fontSize={14}
    >
       - 
    </span>
    <span
      className="sc-htpNat ivUPnB"
      fontSize={14}
    >
      Ruby
    </span>
  </div>
</div>
`;
